jpype/_jproxy.py

```diff
@@ -23,41 +23,47 @@
 # so we can properly handle name mangling on the override.
 
 
+
 def _createJProxy(cls, *intf, **kwargs):
     """ (internal) Create a proxy from a python class with
     @JOverride notation on methods.
     """
-    # Convert the interfaces list
-    actualIntf = _convertInterfaces(intf)
-
-    # Find all class defined overrides
-    overrides = {}
-    for k, v in cls.__dict__.items():
-        try:
-            attr = object.__getattribute__(v, "__joverride__")
-            overrides[k] = (v, attr)
-        except AttributeError:
-            pass
-
-    # Verify all methods are overriden
-    for interface in actualIntf:
-        for method in interface.class_.getMethods():
-            if method.getModifiers() & 1024 == 0:
-                continue
-            if not str(method.getName()) in overrides:
-                raise NotImplementedError("Interface '%s' requires method '%s' to be implemented." % (
-                    interface.class_.getName(), method.getName()))
-
-    members = {}
-
-    def new(tp, *args, **kwargs):
-        self = _jpype._JProxy.__new__(tp, None, actualIntf)
-        cls.__init__(self, *args, **kwargs)
-        return self
-    members['__new__'] = new
-
-    # Return the augmented class
-    return type("proxy.%s" % cls.__name__, (cls, _jpype._JProxy), members)
+    if _jpype.isStarted():
```

Comment:
Given the conversation at #504 I'm not sure JVM is started is the right test for deciding whether we should be doing interface/override checking. I rather think explicit control is a good thing here.

Reply:
We do interface and override checking regardless.  If it is prior to starting then checking is done at the first create, else we execute it immediately.